src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -61,20 +61,21 @@ public TextSubmissionService(TextSubmissionRepository textSubmissionRepository,
      */
     @Transactional
     public TextSubmission handleTextSubmission(TextSubmission textSubmission, TextExercise textExercise, Principal principal) {
+        // Don't allow submissions after the due date (except if the exercise was started after the due date)
+        final var dueDate = textExercise.getDueDate();
+        final var optionalParticipation = participationService.findOneByExerciseIdAndStudentLoginAnyState(textExercise.getId(), principal.getName());
+        if (optionalParticipation.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.FAILED_DEPENDENCY, "No participation found for " + principal.getName() + " in exercise " + textExercise.getId());
+        }
+        final var participation = optionalParticipation.get();
+        if (dueDate != null && participation.getInitializationDate().isAfter(dueDate) && dueDate.isBefore(ZonedDateTime.now())) {
```

Comment:
Instead of `participation.getInitializationDate().isAfter(dueDate)` we should have `participation.getInitializationDate().isBefore(dueDate)` to allow late submission if the participation started after exercise due date

Reply:
That's exactly what gets checked here. If we would change this to `participation.getInitializationDate().isBefore(dueDate)`, we would get an exception if the exercise was started after the due date

Reply:
True!

Reply:
True!